lib/rules/no-unused-vars.js

```diff
@@ -860,7 +1465,18 @@ module.exports = {
                             messageId: "unusedVar",
                             data: unusedVar.references.some(ref => ref.isWrite())
                                 ? getAssignedMessageData(unusedVar)
-                                : getDefinedMessageData(unusedVar)
+                                : getDefinedMessageData(unusedVar),
+                            suggest: [
+                                {
+                                    messageId: "removeVar",
+                                    data: {
+                                        varName: unusedVar.name
+                                    },
+                                    fix(fixer) {
+                                        return handleFixes(fixer, unusedVar, writeReferences);
```

Comment:
`writeReferences` contains only write references in the same execution context (same `variableScope`, e.g., same function):

https://github.com/eslint/eslint/blob/4f08332ac03e51002f8de6da5c5a362608205437/lib/rules/no-unused-vars.js#L850



If we don't want to remove a variable that has references, then in this case it wouldn't work as intended:

```js
let x; // suggestion removes this

() => x = 1;

```